stwcs/tests/test_astrometrydb.py

```diff
@@ -146,7 +143,29 @@ def test_success_offsets(self):
 
 
 def test_db_connection():
-
     adb = astrometry_utils.AstrometryDB()
     assert adb.available
     del adb
+
+
+def test_db_raise_errors_user_override():
+    # should not raise an exception since raise_errors is False
+    db = astrometry_utils.AstrometryDB(url="bad_link/", raise_errors=False)
+    assert db.available is False
+    del db
+
+
+@pytest.mark.xfail
```

Comment:
Is this test not complete? I wonder why it's xfailed and not catching an error.

Reply:
I now use pytest.raises instead of pytest.mark.xfail. I've also added a testing attribute to the class which is meant to speed up the tests that are slowed by the retry wait time.